lib/services/access_request.go

```diff
@@ -1668,51 +1668,52 @@ func roleAllowsResource(
 type ListResourcesRequestOption func(*proto.ListResourcesRequest)
 
 func GetResourceDetails(ctx context.Context, clusterName string, lister ResourceLister, ids []types.ResourceID) (map[string]types.ResourceDetails, error) {
-	var nodeIDs []types.ResourceID
+	var resourceIDs []types.ResourceID
 	for _, resourceID := range ids {
-		if resourceID.Kind != types.KindNode {
-			// The only detail we want, for now, is the server hostname, so we
-			// can skip all other resource kinds as a minor optimization.
-			continue
+		// We're interested in hostname or friendly name details. These apply to
+		// nodes, app servers, and user groups.
+		switch resourceID.Kind {
+		case types.KindNode, types.KindApp, types.KindUserGroup:
+			resourceIDs = append(resourceIDs, resourceID)
 		}
-		nodeIDs = append(nodeIDs, resourceID)
 	}
 
 	withExtraRoles := func(req *proto.ListResourcesRequest) {
 		req.UseSearchAsRoles = true
 		req.UsePreviewAsRoles = true
 	}
 
-	resources, err := GetResourcesByResourceIDs(ctx, lister, nodeIDs, withExtraRoles)
+	resources, err := GetResourcesByResourceIDs(ctx, lister, resourceIDs, withExtraRoles)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
 	result := make(map[string]types.ResourceDetails)
 	for _, resource := range resources {
-		hn, ok := resource.(interface{ GetHostname() string })
-		if !ok {
+		friendlyName := FriendlyName(resource)
+
+		// No friendly name was found, so skip to the next resource.
+		if friendlyName == "" {
 			continue
 		}
+
 		id := types.ResourceID{
 			ClusterName: clusterName,
 			Kind:        resource.GetKind(),
 			Name:        resource.GetName(),
 		}
 		result[types.ResourceIDToString(id)] = types.ResourceDetails{
-			Hostname: hn.GetHostname(),
+			FriendlyName: friendlyName,
 		}
 	}
 
 	return result, nil
 }
 
-func GetNodeResourceIDsByCluster(r types.AccessRequest) map[string][]types.ResourceID {
+// GetResourceIDsByCluster will return resource IDs grouped by cluster.
+func GetResourceIDsByCluster(r types.AccessRequest) map[string][]types.ResourceID {
 	resourceIDsByCluster := make(map[string][]types.ResourceID)
 	for _, resourceID := range r.GetRequestedResourceIDs() {
-		if resourceID.Kind != types.KindNode {
-			continue
-		}
```

Comment:
should this have the same inverse check has line 1676?

Reply:
I elected not to do it here because it has the same net output and is a bit more generic. So this is intentionally not done here.